core/src/main/java/org/axonframework/eventsourcing/eventstore/FilteringDomainEventStream.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2010-2017. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.eventsourcing.eventstore;
+
+import java.util.function.Predicate;
+
+import org.axonframework.eventsourcing.DomainEventMessage;
+
+/**
+ * DomainEventStream implementation that filters a stream.
+ * 
+ * @since 3.2.2
```

Comment:
Nit: It is up to you, but you're free to add the `@author` tag in here for some recognition.